tests/core/test_model.py

```diff
@@ -347,6 +348,33 @@ def test_should_retrain(
     assert retrain.should_retrain_nlu() == fingerprint["retrain_nlu"]
 
 
+async def test_should_not_retrain_core(default_domain_path: Text, tmp_path: Path):
+    # Don't use `default_stories_file` as checkpoints currently break fingerprinting
+    story_file = tmp_path / "simple_story.yml"
+    story_file.write_text(
+        """
+stories:
+- story: test_story
+  steps:
+  - intent: greet
+  - action: utter_greet
+    """
+    )
+    trained_model = await train_core_async(
```

Comment:
I wonder if this test should be in `test_train` as it calls the train methods?

Reply:
Will move it! I came for a fingerprinting perspective that's why I added it here

Reply:
Actually I'll keep it there. It tests `should_retrain` which is part of `model`. the training is just preparing the test data.